daemon/graphdriver/overlay2/check.go

```diff
@@ -126,6 +127,17 @@ func doesSupportNativeDiff(d string) error {
 // metacopy. Nonetheless, a user or kernel distributor may enable metacopy, so
 // we should report in the daemon whether or not we detect its use.
 func usingMetacopy(d string) (bool, error) {
+	userxattr := false
+	if userns.RunningInUserNS() {
+		needed, err := overlayutils.NeedsUserXAttr(d)
```

Comment:
Note for reviewers: this results in us doing a mount to check if we need `userxattr` twice -- however, I didn't want to attach a major refactor to this PR.

The cost is negligible as this is only done at driver `Init`. There will be a follow-up PR that restructures this to avoid the duplicate work.